imports/plugins/core/orders/client/containers/ordersListContainer.js

```diff
@@ -140,30 +162,347 @@ class OrdersListContainer extends Component {
+        } else {
+          Meteor.call("orders/updateHistory", order._id, "Shipping state set by bulk operation", status);
+        }
+        orderCount++;
+        if (orderCount === selectedOrders.length) {
+          this.setState({
+            shipping: {
+              [status]: true
+            },
+            isLoading: {
+              [status]: false
+            }
+          });
+          Alerts.alert({
+            text: i18next.t("order.orderSetToState", { orderNumber: selectedOrders.length, orderText: orderText, status: status }),
+            type: "success",
+            allowOutsideClick: false
+          });
+        }
+      });
+    });
+  }
+
+  displayOrderText = (selectedOrders) => {
+    let orderText = "";
+    if (selectedOrders.length > 1) {
+      orderText = "Orders have";
+    } else {
+      orderText = "Order has";
+    }
+
+    return orderText;
+  }
+
+  displayAlert = (selectedOrders, status, whichFalseState, falsePreviousStatuses, falseCurrentState, trueCurrentState) => {
+    const capitalizeStatus = status[0].toUpperCase() + status.substr(1).toLowerCase();
+    let orderText = "";
+    let skippedOrdersText = "";
+
+    if (selectedOrders.length > 1) {
+      orderText = "orders";
+    } else {
+      orderText = "order";
+    }
+
+    if (falsePreviousStatuses > 1) {
+      skippedOrdersText = "are";
+    } else {
+      skippedOrdersText = "is";
+    }
+
+    // if the order(s) want to skip the previous states, display alert
+    if (falsePreviousStatuses) {
+      Alerts.alert({
+        text: i18next.t("order.skippedBulkOrdersAlert", {
+          selectedOrders: selectedOrders.length, orderText: orderText, status: capitalizeStatus,
+          numberOfSkippedOrders: falsePreviousStatuses, skippedOrdersText: skippedOrdersText, skippedState: whichFalseState
+        }),
+        type: "warning",
+        showCancelButton: true,
+        showCloseButton: true,
+        allowOutsideClick: false,
+        confirmButtonText: i18next.t("order.approveBulkOrderAction"),
+        cancelButtonText: i18next.t("order.cancelBulkOrderAction")
+      }, (setSelected) => {
+        if (setSelected) {
+          // set status of order(s) if this action is confirmed
+          this.shippingStatusUpdateCall(selectedOrders, status);
+        }
+      });
+      // if the order(s) are following proper flow, set the status
+    } else if (!falsePreviousStatuses && falseCurrentState) {
+      this.shippingStatusUpdateCall(selectedOrders, status);
+      // display alert if order(s) are already in this state
+    } else if (!falsePreviousStatuses && !falseCurrentState && trueCurrentState) {
+      Alerts.alert({
+        text: i18next.t("order.orderAlreadyInState", { orderText: this.displayOrderText(selectedOrders), status: status })
+      });
+    }
+  }
+
+  displayRegressionAlert = (selectedOrders, ordersToRegress, status, whichFalseState, falsePreviousStatuses, falseCurrentState, trueCurrentState) => {
```

Comment:
There's a bunch of places where the line is too long. Your editor should be restricting you to 120 chars.